pkg/detectors/uri/uri.go

```diff
@@ -23,7 +21,9 @@ type Scanner struct {
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	keyPat = regexp.MustCompile(`\b[a-zA-Z]{1,10}:?\/\/[-.%\w{}]{1,50}:([-.%\S]{3,50})@[-.%\w\/:]+\b`)
+	keyPat = regexp.MustCompile(`\b(?:http(?:s)?:)?\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
+
+	client = common.SaneHttpClient()
 )
 
 type proxyRes struct {
```

Comment:
nit: I know you didn't touch this, but looks like we no longer use this struct. We can probably remove?